contracts/transfer/src/transfer.rs

```diff
@@ -135,6 +135,25 @@ impl TransferContract {
             Ok(t || self.nullifiers.get(n).map(|n| n.is_some())?)
         })
     }
+
+    /// Takes a slice of nullifiers and returns a vector containing the ones
+    /// that already exists in the contract
+    pub fn find_existing_nullifiers(
+        &self,
+        nullifiers: &[BlsScalar],
+    ) -> Result<Vec<BlsScalar>, Error> {
+        nullifiers
```

Comment:
Maybe leave an open issue that this can be optimized in the future?

Reply:
How do you suggest to optimize? I also tried as API returning an iterator instead of collecting the values in a vec, however in doing so you lost the capability to return the error as group of course (it will be part of the single element of the iterator), and the usage of the API will be less friendly; but if we're fine with that I already have the code.